Create Plugin fails tsc build step

Running yarn create-plugin causes failure at providers[env] step in EnvironmentHandler.

```
$ yarn create-plugin                             [13:57:57]
yarn run v1.22.1
$ backstage-cli create-plugin
? Enter an ID for the plugin [required] foobar
? Enter the owner(s) of the plugin. If specified, this will be added to CODEOWNERS for the plugi
n path. [optional] 

Creating the plugin...

 Checking if the plugin ID is available:
  checking      foobar ✔ 

 Creating a temporary plugin directory:
  creating      temporary directory ✔ 

 Preparing files:
  copying       .eslintrc.js ✔ 
  templating    README.md.hbs ✔ 
  templating    package.json.hbs ✔ 
  copying       index.tsx ✔ 
  templating    plugin.test.ts.hbs ✔ 
  copying       index.ts ✔ 
  templating    plugin.ts.hbs ✔ 
  copying       setupTests.ts ✔ 
  templating    ExampleComponent.test.tsx.hbs ✔ 
  templating    ExampleComponent.tsx.hbs ✔ 
  copying       index.ts ✔ 
  templating    ExampleFetchComponent.test.tsx.hbs ✔ 
  templating    ExampleFetchComponent.tsx.hbs ✔ 
  copying       index.ts ✔ 

 Moving to final location:
  moving        foobar ✔ 

 Building the plugin:
  executing     yarn install ✔ 
  executing     yarn tsc ◜ error Command failed with exit code 2.
$ /Users/turista/roku/backstage/node_modules/.bin/tsc
../../packages/cli/src/lib/run.ts:70:21 - error TS2345: Argument of type 'ChildProcess' is not assignable to parameter of type 'ChildProcess & { exitCode?: number | undefined; }'.
  Type 'ChildProcess' is not assignable to type '{ exitCode?: number | undefined; }'.
    Types of property 'exitCode' are incompatible.
      Type 'number | null' is not assignable to type 'number | undefined'.
        Type 'null' is not assignable to type 'number | undefined'.

70   await waitForExit(child, name);
                       ~~~~~

../auth-backend/src/lib/EnvironmentHandler.ts:36:39 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string | number | symbol'.
  Type 'undefined' is not assignable to type 'string | number | symbol'.

36     if (this.providers.hasOwnProperty(env)) {
                                         ~~~

../auth-backend/src/lib/EnvironmentHandler.ts:37:29 - error TS2538: Type 'undefined' cannot be used as an index type.

37       return this.providers[env];
                               ~~~


Found 3 errors.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  executing     yarn tsc ✖ 

Could not execute command yarn tsc

It seems that something went wrong when creating the plugin 🤔
We are going to clean up, and then you can try again.

 Cleanup:
  remove        temporary directory ✔ 

🔥  Failed to create plugin!

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
## Expected Behavior

It creates plugin and ts-check succeeds

## Current Behavior

It doesn't

## Possible Solution

fix the typing or add a default value to env variable in 
`auth-backend/src/lib/EnvironmentHandler.ts`

## Steps to Reproduce

1. Run `yarn create-plugin`
2. give it foobar name

## Context
Trying to demo the features for creating a plugin, but it doesn't work.

## Your Environment

- NodeJS Version (v12):
- Mac OS 10.15.4

--- a/packages/cli/e2e-test/cli-e2e-test.js
+++ b/packages/cli/e2e-test/cli-e2e-test.js
@@ -201,7 +201,7 @@ async function createPlugin(pluginName, appDir) {
     await waitForExit(child);
 
     const pluginDir = resolvePath(appDir, 'plugins', pluginName);
-    for (const cmd of [['lint'], ['test', '--no-watch']]) {
+    for (const cmd of [['tsc'], ['lint'], ['test', '--no-watch']]) {
       print(`Running 'yarn ${cmd.join(' ')}' in newly created plugin`);
       await runPlain(['yarn', ...cmd], { cwd: pluginDir });
     }

--- a/packages/cli/src/lib/run.ts
+++ b/packages/cli/src/lib/run.ts
@@ -92,7 +92,7 @@ export async function runCheck(cmd: string, ...args: string[]) {
 }
 
 export async function waitForExit(
-  child: ChildProcess & { exitCode?: number },
+  child: ChildProcess & { exitCode: number | null },
   name?: string,
 ): Promise<void> {
   if (typeof child.exitCode === 'number') {

--- a/plugins/auth-backend/src/lib/EnvironmentHandler.ts
+++ b/plugins/auth-backend/src/lib/EnvironmentHandler.ts
@@ -33,7 +33,7 @@ export class EnvironmentHandler implements AuthProviderRouteHandlers {
   ): AuthProviderRouteHandlers | undefined {
     const env = req.query.env?.toString();
 
-    if (this.providers.hasOwnProperty(env)) {
+    if (env && this.providers.hasOwnProperty(env)) {
       return this.providers[env];
     }
 

